### PR TITLE
Merge timeline panels

### PR DESCRIPTION
**Pull request by flenter**
This PR also:
* adds stack traces with vscode (file) links

https://github.com/user-attachments/assets/55f542c7-473b-4eb6-8e03-7032dae7e6d4

### CODE REVIEW

File: packages/client-library-otel/src/instrumentation.ts [hunk 1]
```diff
@@ -275,7 +276,13 @@ function setupTracerProvider(options: {
   const exporter = new OTLPTraceExporter({
     url: options.endpoint,
   });
-  provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+  provider.addSpanProcessor(
+    new SimpleSpanProcessor(exporter),
```
Comment (flenter) [inline packages/client-library-otel/src/instrumentation.ts:280]: Outside of the scope of this pr, but when generating a large amount of traces, they get lost (aren't sent). Switching over to the BatchSpanProcessor fixes that partially, but now traces aren't always sent (immediately?)

**Review by mies — APPROVED** [review]

**Comment by mies** [comment]
timestamps are a bit funky on 14" mbpro in Arc Browser

CleanShot 2024-10-04 at 06 48 26@2x

**Review by stephlow — APPROVED** [review]

**Comment by brettimus** [comment]
alignment / overflow funkiness for me (see request status + url, they're overflowing)

image

File: studio/src/components/Timeline/DetailsList/spans/FetchSpan/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+/**
+ * Returns a component and title for a vendor-specific span.
+ * @param span The span to render.
+ * @returns A component and title for a vendor-specific section of the span.
+ */
+export function useVendorSpecificSection(vendorInfo: VendorInfo) {
+  return useMemo(() => {
+    // const vendorInfo =
```
Comment (brettimus) [inline studio/src/components/Timeline/DetailsList/spans/FetchSpan/hooks.tsx]:
```suggestion
```

File: studio/src/components/Timeline/DetailsList/spans/GenericSpan/GenericSpanContent.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            args={exception.args}
+            level={exception.level}
+            message={exception.message}
+            callerLocations={exception.callerLocations}
+            service={exception.service}
+          />
+        </SubSection>
+      )}
+      <SubSection className="gap-0">
+        <SubSectionHeading>Meta data</SubSectionHeading>
```
Comment (brettimus) [inline studio/src/components/Timeline/DetailsList/spans/GenericSpan/GenericSpanContent.tsx]:
```suggestion
        <SubSectionHeading>Metadata</SubSectionHeading>
```

File: studio/src/pages/RequestorPage/ai/AiTestGenerationDrawer.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ export function AiTestGenerationDrawer({
       <DrawerContent className="py-4">
         <div className="mx-auto w-full max-w-md">
           <DrawerHeader className="mt-2 p-2">
-            <DrawerTitle>Close the Loop</DrawerTitle>
+            <DrawerTitle>Close the Loop 2</DrawerTitle>
```
Comment (brettimus) [inline studio/src/pages/RequestorPage/ai/AiTestGenerationDrawer.tsx]:
```suggestion
            <DrawerTitle>Close the Loop</DrawerTitle>
```

File: studio/src/utils/index.ts [hunk 5]
```diff
@@ -195,7 +195,8 @@ export const safeParseJson = (jsonString: string) => {
     return parsed;
   } catch (error) {
     console.error("Failed to parse JSON:", error);
-    return jsonString;
+    // return jsonString;
```
Comment (brettimus) [inline studio/src/utils/index.ts]:
```suggestion
```

File: studio/src/utils/otel-helpers.ts [hunk 6]
```diff
@@ -28,19 +28,19 @@ const isErrorEvent = (event: OtelEvent) => {
...
   return span.events?.some(isErrorEvent);
 };
 
-export const getErrorEvents = (span: OtelSpan) => {
+export const getErrorEvents = (span: Pick<OtelSpan, "events">) => {
   return span.events?.filter(isErrorEvent) ?? [];
 };
 
-export function isFpxRequestSpan(span: OtelSpan) {
+export function isFpxRequestSpan(span: Pick<OtelSpan, "name" | "kind">) {
```
Comment (brettimus) [inline studio/src/utils/otel-helpers.ts]: not sure i like the use of `Pick` here to be honest. these functions are discriminators on the OtelSpan type, and so i think they should only work on data that conform to that type.

imo a plain-old javascipt object with "kind" and "name" attrs would return true from this, which i don't think is expected behavior

**Comment by brettimus** [comment]
maybe smaller rounded corners? 
image


this is what it's like with `rounded-lg` instead

image


file: fpx/studio/src/pages/RequestorPage/LogsTable/LogsTableRow/LogRow.tsx

```tsx
          className={cn(
            "cursor-pointer focus:outline-none focus:ring-1 focus:ring-blue-500 focus:ring-inset",
            isExpanded ? "rounded-t-lg" : "rounded-lg",
          )}
```

File: studio/src/components/Timeline/DetailsList/TimelineDetailsList/TimelineListDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                duration={duration}
+                indent={0}
+                withLogs={withLogs}
+              />
+            )
+          : waterfall.map((item) => {
+              const isLog = isMizuOrphanLog(item);
+              if (isLog) {
+                console.log("log", item);
+              }
```
Comment (brettimus) [inline studio/src/components/Timeline/DetailsList/TimelineDetailsList/TimelineListDetails.tsx:76]:
```suggestion
```

**Review by brettimus — APPROVED** [review]
just need to tidy up some of the stuff with 

- overflow (extra margin appearing on top-level request entry)
- timestamps

**Comment by flenter** [comment]
> alignment / overflow funkiness for me (see request status + url, they're overflowing)
> 
> image

I think this was an issue on smaller screens. Fixed it

**Pull request merged**